base/channels.jl

```diff
@@ -26,10 +26,21 @@ type Channel{T} <: AbstractChannel
         end
         new(Condition(), Condition(), :open, Array{T}(0), sz, Array{Condition}(0))
     end
+
+    # deprecated empty constructor
+    function Channel()
+        depwarn(string("The empty constructor Channel() is deprecated. ",
```

Comment:
why can't it be an outer constructor? especially with two different signatures in different places that should be removed, those comments are going to be easy to miss